define/types.go

```diff
@@ -101,11 +101,43 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
+	Type     BuildOutputType
+	Attrs    map[string]string
 	Path     string // Only valid if !IsStdout
 	IsDir    bool
 	IsStdout bool
 }
 
+type BuildOutputType int
+
+const (
+	Docker BuildOutputType = iota
```

Comment:
Having the zero value be `BuildOutputTypeUninitialized` or `…Undefined` would add a bit of protection against coding errors.